app/routes/user.py

```diff
@@ -519,4 +522,30 @@ def user_signup_fake():
   db.session.add(user)
   db.session.commit()
   user.id
-  return user.get_dict()
\ No newline at end of file
+  return user.get_dict()
+
+@bp.route('/<int:id>/upload', methods=['POST'])
+@jwt_required()
+def upload(id):
+  if 'file' not in request.files:
+    return { 'msg': 'No file included in request' }, 422
+  # If the user does not select a file, the browser submits an
+  # empty file without a filename.
+  file = request.files.get('file')
+  if file.filename == '':
+      return { 'msg': 'Submitted an empty file' }, 422
+  import uuid
+  s3_key = str(get_current_user().id) + '_' + str(uuid.uuid4())
```

Comment:
for the user profile pic we should save the file name just as their username if they have one. that way if they change their profile pic we just overwrite the old one.

if not we could fallback to this